Design/Instrument-2.0/ScanningInstruments_Design.md

```diff
@@ -0,0 +1,102 @@
+
+This can be used to access methods on `DetectorInfo`, for example `m_detectorInfo.twoTheta(detIndex, timeIndex)`.
+
+#### `MatrixWorkspace` & `ISpectrum`/`Spectrum`
+
+As metioned under **`SpectrumInfo`** `MatrixWorkspace` will require a `getTimeIndexedDetectorIDs(index)` method.
+
+This implies the member variable `std::set<detid_t> detectorIDs` in `ISpectrum` will need to become `std::set<std::pair<detid_t, size_t>> timeIndexedDetectorIDs`.
+
+The corresponding methods in `ISpectrum` would also need to be replicated for the time indexed case. For example `addDetectorID(const std::pair<detid_t, size_t> timeIndexedDetID)`.
+
+The original methods such as `this->detectorIDs.insert(detID)` should remain and add detectors at time index 0. This allows existing algorithms, e.g. loaders, that do not need time indexing to work as before. Similarly the `getDetectorIDs(index)` method would still return detector IDs without the time indexing information.
+
+### Implementation
+
+For more details on the rollout plan see the document [Scanning Instruments Plan](ScanningInstruments_Plan.md).
+
+For this design to be implemented the use of `SpectrumInfo` and `DetectorInfo` will need to be rolled out in Mantid. The rollout for [`SpectrumInfo`](https://github.com/mantidproject/mantid/issues/17743) has been started.
+
+### New DetectorInfo Signatures
+
+Currently detector info public methods look like the following.
+
+```cpp
+DetectorInfo(boost::shared_ptr<const Geometry::Instrument> instrument,
+           Geometry::ParameterMap *pmap = nullptr);
+
+bool isMonitor(const size_t index) const;
+bool isMasked(const size_t index) const;
+double l2(const size_t index) const;
+double twoTheta(const size_t index) const;
+double signedTwoTheta(const size_t index) const;
+Kernel::V3D position(const size_t index) const;
+Kernel::Quat rotation(const size_t index) const;
+
+void setPosition(const size_t index, const Kernel::V3D &position);
+void setRotation(const size_t index, const Kernel::Quat &rotation);
+
+void setPosition(const Geometry::IComponent &comp, const Kernel::V3D &pos);
+void setRotation(const Geometry::IComponent &comp, const Kernel::Quat &rot);
+```
+
+The following methods will be given require a time index access, in addition to the detector index.
+
+```cpp
+DetectorInfo(boost::shared_ptr<const Geometry::Instrument> instrument,
+           Geometry::ParameterMap *pmap = nullptr);
+
+bool isMonitor(const size_t index) const;
+bool isMasked(const size_t index, const size_t timeIndex=0) const;
+double l2(const size_t index, const size_t timeIndex=0) const;
+double twoTheta(const size_t index, const size_t timeIndex=0) const;
+double signedTwoTheta(const size_t index, const size_t timeIndex=0) const;
+Kernel::V3D position(const size_t index, const size_t timeIndex=0) const;
+Kernel::Quat rotation(const size_t index, const size_t timeIndex=0) const;
+
+void setPosition(const size_t index, const Kernel::V3D &position, const size_t timeIndex=0);
```

Comment:
Yes, I think this could make sense. So when creating the `SpectrumInfo` object you would need to pass a vector of std::set<std::pair<detid_t, size_t>> objects, for step scans?